automation/src/test/scala/org/broadinstitute/dsde/test/api/SnapshotAPISpec.scala

```diff
@@ -153,6 +160,104 @@ class SnapshotAPISpec extends AnyFreeSpecLike with Matchers with BeforeAndAfterA
+          // TODO: a function createMethodConfigInWorkspace exists in workbench-libs, but not flexible/updated to accept dataReferenceName and maybe other parameters
+          // TODO: other ways to clean up payload building process in this test? move to a private method within this test class?
+          val createMethodConfigPayload = Map("methodRepoMethod" -> Map("methodUri" -> "agora://gatk/echo_to_file/9","methodName" -> "echo_to_file","methodNamespace" -> "gatk","methodVersion" -> 9),
+            "name" -> "echo_to_file-configured",
+            "namespace" -> "gatk",
+            "rootEntityType" -> "vcf_file",
+            "prerequisites" -> Map(),
+            "inputs" -> Map("echo_strings.echo_to_file.input1" -> "this.VCF_File_Name"),
+            "outputs" -> Map("echo_strings.echo_to_file.out" -> "workspace.output"),
+            "methodConfigVersion" -> 1,
+            "deleted" -> false,
+            "dataReferenceName" -> "testSnapshotForRunningAnalysis"
+          )
+          Rawls.postRequest(
+            uri = createMethodConfigUrl.toString(),
+            content = createMethodConfigPayload)
+
+          // run analysis on the snapshot
+          val createSubmissionUrl  = Uri(Rawls.url).withPath(Path(s"/api/workspaces/$projectName/$workspaceName/submissions"))
+          val createSubmissionPayload = Map(
+            "useCallCache" -> true,
+            "deleteIntermediateOutputFiles" -> false,
+            "methodConfigurationNamespace" -> "gatk",
+            "methodConfigurationName" -> "echo_to_file-configured"
+          )
+          val response = Rawls.postRequest(
+            uri = createSubmissionUrl.toString(),
+            content = createSubmissionPayload)
+          // TODO: revisit the mapper usage here
+          val submissionId = mapper.readTree(response).get("submissionId").asText()
+
+          // wait for submission to complete
+          Submission.waitUntilSubmissionComplete(projectName, workspaceName, submissionId)
+
+          // verify submission status is done
+          val expectedSubmissionStatus = "Done"
+          val actualSubmissionStatus = Submission.getSubmissionStatus(projectName, workspaceName, submissionId)
+          withClue(s"Submission $projectName/$workspaceName/$submissionId status should be $expectedSubmissionStatus") {
+            actualSubmissionStatus shouldBe expectedSubmissionStatus
+          }
+
+          // verify workflows succeeded
+          //TODO: does this step make sense?
+          val getSubmissionUrl  = Uri(Rawls.url).withPath(Path(s"/api/workspaces/$projectName/$workspaceName/submissions/$submissionId"))
+          val submissionResponse = Rawls.parseResponse(Rawls.getRequest(uri = getSubmissionUrl.toString))
+          // TODO: revisit the mapper usage here
+          val workflows: List[JsonNode] = mapper.readTree(submissionResponse).get("workflows").elements().asScala.toList
```

Comment:
same here - parse into case class, not `JsonNode`

Reply:
Created AS-622 to keep track of this.